### PR TITLE
Upgrade to Pint 0.7

### PR DESCRIPTION
**Pull request by noahl**
Now that Pint 0.7 has been released, change our code to fix some
incompatibilities with 0.7 and require pint>=0.7 in requirements.txt.

### CODE REVIEW

**Comment by noahl** [comment]
@stfeng @yuyantingzero please take a look.

For some reason this has been a day full of upgraded components causing incompatibility. I have two more pending PRs that depend on this fix, both updating components.

File: perfkitbenchmarker/__init__.py [hunk 1]
```diff
@@ -21,28 +21,21 @@
 # we create it here.
 UNIT_REGISTRY = pint.UnitRegistry()
 
-# Pint 0.6 uses 'Bo' as the abbreviation for a byte. We want to use
-# 'B', like the rest of the world.
-UNIT_REGISTRY.define('byte = 8 * bit = B')
```
Comment (yuyantingzero) [inline perfkitbenchmarker/__init__.py]: Does pint 0.7 recognize byte and B?

**Comment by yuyantingzero** [comment]
LGTM, one small comment

File: perfkitbenchmarker/__init__.py (hunk 1 above)
Comment (noahl) [inline perfkitbenchmarker/__init__.py]: Yes, that's why I removed the redefinition.

**Comment by noahl** [comment]
@yuyantingzero Thanks for the quick review!

**Pull request merged**